test.sh

```diff
@@ -1,40 +1,74 @@
-#!/bin/bash
+#!/usr/bin/env bash
+
+# -e Stops execution in the instance of a command or pipeline error
+# -u Treat unset variables as an error and exit immediately
+set -eu
+
 # Run all tests and coverage checks. Called from Travis automatically, also
 # suitable to run manually. See list of prerequisite packages in .travis.yml
 if type realpath >/dev/null 2>&1 ; then
-  cd $(realpath $(dirname $0))
+  cd "$(realpath -- $(dirname -- "$0"))"
 fi
 
-set -exo pipefail
-TRAVIS=${TRAVIS:-false}
+#
+# Defaults
+#
+STATUS="FAILURE"
+# RUN, UNIT_FILTER, and INT_FILTER vars are here for backwards compatibility,
+# and should be set to () unconditionally in the future.
+if [ -z "${RUN+x}" ]; then RUN=(); fi
+if [ -z "${UNIT_FILTER+x}" ]; then UNIT_FILTER=(); fi
+if [ -z "${INT_FILTER+x}" ]; then INT_FILTER=(); fi
+ENABLE_UNIT_RACE="false"
+export BOULDER_CONFIG_DIR="${BOULDER_CONFIG_DIR:-"test/config"}"
 
-# The list of segments to run. To run only some of these segments, pre-set the
-# RUN variable with the ones you want (see .travis.yml for an example).
-# Order doesn't matter. Note: gomod-vendor is specifically left out of the
-# defaults, because we don't want to run it locally (it could delete local
-# state) We also omit coverage by default on local runs because it generates
-# artifacts on disk that aren't needed.
-RUN=${RUN:-lints unit integration}
+#
+# Print Functions
+#
+function print_outcome() {
+  if [ "$STATUS" == SUCCESS ]
+  then
+    echo -e "\e[32m"$STATUS"\e[0m"
+  else
+    echo -e "\e[31m"$STATUS"\e[0m"
+  fi
+}
+
+function print_list_of_integration_tests() {
+  for file in ./test/integration/*.go; do
+    [ -e "$file" ] || continue
```

Comment:
When would a file not exist after being listed in the glob above?

Reply:
If there are no `*.go` files in the directory, then the for loop is executed once, with `file="*.go"`. The workaround is to test whether there is a matching file.